pkg/cluster/deploybaseresources_additional.go

```diff
@@ -92,6 +92,20 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
+func (m *manager) fpspStorageBlobContributorRBAC(storageAccountName string) (*arm.Resource, error) {
```

Comment:
IMO make the `StorageBlobContributorRBAC` a function that takes `storageAccountName` and `principal_id` and just pass the principal ID to it, incase we want to reuse this in the future.

Reply:
The function now takes the principalID